src/test/kotlin/io/kotest/extensions/spring/security/SpringMockUserExtensionTest.kt

```diff
@@ -0,0 +1,141 @@
+package io.kotest.extensions.spring.security
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.descriptors.append
+import io.kotest.core.descriptors.toDescriptor
+import io.kotest.core.names.TestName
+import io.kotest.core.source.sourceRef
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import io.kotest.core.test.TestType
+import io.kotest.extensions.spring.SpringExtension
+import io.kotest.matchers.collections.shouldContainOnly
+import io.kotest.matchers.equals.shouldBeEqual
+import io.kotest.matchers.shouldBe
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.test.context.TestSecurityContextHolder
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken
+
+class SpringMockUserExtensionTest : FunSpec() {
+
+   override fun extensions() = listOf(SpringExtension)
+
+   init {
+
+      afterAny {
+         SecurityContextHolder.getContext().authentication = null
+      }
+
+      test("should set user before Spring in Spring security context") {
+         SpringMockUserExtension().beforeSpring(
+            TestCase(
+               descriptor = SpringMockUserExtensionTest::class.toDescriptor().append("aaa"),
+               name = TestName("name"),
+               spec = this@SpringMockUserExtensionTest,
+               test = {},
+               source = sourceRef(),
+               type = TestType.Test
+            )
```

Comment:
Could perhaps be extracted, seems to be used in every test w/o any relevant differences?